lambdas/handlers/mns_notification_handler.py

```diff
@@ -0,0 +1,44 @@
+import json
+
+from enums.mns_notification_types import MNSNotificationTypes
+from models.mns_sqs_message import MNSSQSMessage
+from services.process_mns_message_service import MNSNotificationService
+from utils.audit_logging_setup import LoggingService
+from utils.decorators.ensure_env_var import ensure_environment_variables
+from utils.decorators.handle_lambda_exceptions import handle_lambda_exceptions
+from utils.decorators.override_error_check import override_error_check
+from utils.decorators.set_audit_arg import set_request_context_for_logging
+
+logger = LoggingService(__name__)
+
+
+@set_request_context_for_logging
+@ensure_environment_variables(
+    names=[
+        "APPCONFIG_CONFIGURATION",
+        "APPCONFIG_ENVIRONMENT",
+        "LLOYD_GEORGE_DYNAMODB_NAME",
+        "MNS_NOTIFICATION_QUEUE_URL",
+    ]
+)
+@override_error_check
+@handle_lambda_exceptions
+def lambda_handler(event, context):
+    logger.info(f"Received MNS notification event: {event}")
+
+    sqs_messages = event.get("Records", [])
+
+    for sqs_message in sqs_messages:
+        try:
+            sqs_message = json.loads(sqs_message["body"])
+
+            mns_message = MNSSQSMessage(**sqs_message)
+            MNSSQSMessage.model_validate(mns_message)
+
+            if mns_message.type in MNSNotificationTypes.__members__.values():
+                notification_service = MNSNotificationService()
+                notification_service.handle_mns_notification(mns_message)
+                logger.info("Continuing to next message.")
+        except Exception as error:
```

Comment:
Do we know what specific exception can be raised here?

Reply:
There are a few different types that could bubble up and just wanted to make sure they are caught. Is there a better way to handle this?

Reply:
Thought about this, there could be a pydantic validation error here that doesn't get handled, something like malformed message we have else where in the code base..?

Reply:
I guess a catch all is required if there's a risk of this falling over, I'm not against the Exception but if we can catch what we know that would be ideal